pandas/core/generic.py

```diff
@@ -190,12 +189,7 @@ class NDFrame(PandasObject, SelectionMixin):
     _metadata = []  # type: List[str]
     _is_copy = None
     _data = None  # type: BlockManager
-
-    if TYPE_CHECKING:
-        # TODO(PY36): replace with _attrs : Dict[Hashable, Any]
-        # We need the TYPE_CHECKING, because _attrs is not a class attribute
-        # and Py35 doesn't support the new syntax.
-        _attrs = {}  # type: Dict[Optional[Hashable], Any]
+    _attrs = {}  # type: Dict[Optional[Hashable], Any]
```

Comment:
```suggestion
    _attrs: Dict[Optional[Hashable], Any]
```

Reply:
Good point, but I'd prefer that change in a separate PR.

Both should be equivalent afaik. And we don't have the new typing annotation anywhere yet. And I'm sure there are many other changes like this one that could be done at this point. My preference would be to make this change together with other changes of this kind, to not mix things.

Reply:
> Both should be equivalent afaik.

not quite. The reason for the TYPE_CHECKING block was to ensure that _attrs is not initialized to an empty dict at runtime.

if using python 3.6 syntax you can declare a type without initialising the value.

so here you either need to revert to include the TYPE_CHECKING or convert to 3.6 syntax